src/geometry_canvas.jl

```diff
@@ -163,15 +163,16 @@ function GeometryCanvas{T}(obj=Observable(_geomtype(T)[]);
     draw!(figure, axis, canvas;
         scatter_kw, lines_kw, poly_kw, current_point_kw, show_current_point
     )
-    addtoswitchers!(canvas)
     add_events!(canvas; mouse_property)
     return canvas
 end
 
 _current_point_obs(::Type{<:Point}) = Observable(1)
 _current_point_obs(::Type) = Observable((1, 1))
 
-_geomtype(T) = T
+_geomtype(T::Type) = T
+_geomtype(::Type{LineString}) = LineString{2,Float64,Point{2,Float64}}
+_geomtype(::Type{Polygon}) = Polygon{2,Float64,Point{2,Float64}}
 _geomtype(::Type{<:Point}) = Point2f
```

Comment:
```suggestion
_geomtype(::Type{<:Point}) = Point2{Float64}
```
Just to switch everything to Float64